pkg/util/timeutil/time_zone_util.go

```diff
@@ -66,3 +67,45 @@ func ParseFixedOffsetTimeZone(location string) (offset int, origRepr string, suc
 	}
 	return offset, strings.TrimSuffix(strings.TrimPrefix(origRepr, "("), ")"), true
 }
+
+// TimeZoneOffsetStringConversion converts a time string to offset seconds
+// Supported time zone strings :- GMT/UTC±[00:00:00 - 15:59:59], GMT/UTC±[0-16]
+// Unsupported time zone strings :- GMT/UTC±16:00 to upper, GMT/UTC±6.5
+// (case insensitive)
+func TimeZoneOffsetStringConversion(s string) (offset int64, ok bool) {
+	pattern := `(?i)(GMT|UTC)[+-]((((0?\d)|(1[0-5])):([0-5]\d))|(((0?\d)|(1[0-5])):([0-5]\d):([0-5]\d))|((1+[0-6])|(0?\d)))\b$`
+	re := regexp.MustCompile(pattern)
```

Comment:
nit: we typically put `MustCompile` as a global var, so it only has to execute once to generate the regex and it will panic at init time rather than runtime if the regex does not compile :)

Reply:
Ok 👍